packages/instant-apps-components/src/components/instant-apps-app-guide/instant-apps-app-guide.tsx

```diff
@@ -21,52 +26,72 @@ export class InstantAppsAppGuide {
   @Prop()
   data: AppGuidePage[];
 
+  @Watch('data')
+  watchPropHandler(newValue: AppGuidePage[]) {
+    this.viewModel.pages = newValue;
+    this.el.dispatchEvent(new CustomEvent('calciteCarouselChange'));
+  }
+
   @State() messages: typeof AppGuide_T9n;
+  @State() headerTitle: string;
 
-  viewModel: AppGuideViewModel = new AppGuideViewModel();
+  private viewModel: AppGuideViewModel = new AppGuideViewModel();
+  private carouselRef: HTMLCalciteCarouselElement;
 
-  componentWillRender() {
-    this._initialize();
+  connectedCallback() {
+    this.viewModel.pages = this?.data || [];
+    this.headerTitle = this?.data[0]?.title;
   }
 
-  componentDidRender() {
+  componentDidLoad() {
     getMessages(this);
+
+    // Listen for the calciteCarouselChange event to update the header title to match the selected page
+    this.el.addEventListener('calciteCarouselChange', (_event: CustomEvent) => {
```

Comment:
This can be added directly in the JSX

```
<calcite-carousel onCalciteCarouselChange={() => console.log("Howdy!")}>
    <calcite-carousel-item></calcite-carousel-item>
</calcite-carousel>
```

Reply:
Done